pkg/controller/node/node_controller.go

```diff
@@ -298,6 +298,7 @@ func NewNodeController(
 		zoneStates:                  make(map[string]ZoneState),
 		runTaintManager:             runTaintManager,
 		useTaintBasedEvictions:      useTaintBasedEvictions && runTaintManager,
+		taintNodeByCondition:        taintNodeByCondition,
```

Comment:
Please add a unit test to ensure this.

Reply:
let me add an integration or e2e test for this :).

Reply:
Thanks, e2e is better.